### PR TITLE
Fix PHP 7 warnings

### PR DESCRIPTION
**Pull request by vianasw**
This PR is to address a couple of warnings we have regarding PHP 7 compatibility reported in this [issue](https://github.com/Automattic/vaultpress/issues/110).

There are just two small changes:

1. Remove call to the deprecated method `get_magic_quotes_gpc`. It was also unnecessary in this context because:

> WordPress ignores the built-in PHP magic quotes setting and the value of get_magic_quotes_gpc() and will always add magic quotes (even after the feature is removed from PHP in 5.4).

https://developer.wordpress.org/reference/functions/stripslashes_deep/#more-information

2. Change func_get_args call order. 
 There was a warning complaining about accessing a function argument after calling `func_get_args`. This can be an issue because:

> Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value.

We weren't modifying the function argument in any way but the warning was there anyway.

### CODE REVIEW

File: vaultpress.php [hunk 1]
```diff
@@ -3046,12 +3046,9 @@ function register_via_jetpack( $already_purchased = false ) {
...
 	if ( !function_exists( 'wp_magic_quotes' ) ) {
-		// If already slashed, strip.
-		if ( get_magic_quotes_gpc() ) {
-			$_GET    = stripslashes_deep( $_GET    );
-			$_POST   = stripslashes_deep( $_POST   );
-			$_COOKIE = stripslashes_deep( $_COOKIE );
-		}
+		$_GET    = stripslashes_deep( $_GET    );
+		$_POST   = stripslashes_deep( $_POST   );
+		$_COOKIE = stripslashes_deep( $_COOKIE );
```
Comment (macbre) [inline vaultpress.php:3051]: PHP manual says that `get_magic_quotes_gpc` now always returns `false` (as the magic quotes  feature has been deprecated for quite a long an was finally removed in PHP8).

Hence, shouldn't we just get rid of these three lines, previously wrapped in an `if` statement?

**Comment by macbre** [comment]
Travis complains:

```
$ composer install
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. It is recommended that you run `composer update` or `composer update <package name>`.
Your lock file does not contain a compatible set of packages. Please run composer update.
```

File: vaultpress.php (hunk 1 above)
Comment (vianasw) [reply vaultpress.php:3051]: Makes sense, thanks for checking.

File: vaultpress.php [hunk 2]
```diff
@@ -3046,12 +3046,9 @@ function register_via_jetpack( $already_purchased = false ) {
...
```
Comment (macbre) [inline vaultpress.php:3052]:
```suggestion
```

Removing this section as per the discussion above.

Comment (macbre) [reply vaultpress.php:3052]: https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/load.php#L1049

**Comment by macbre** [comment]
Closing in favour of #123

**Pull request closed without merging**